src/include/serial_inline.h

```diff
@@ -224,11 +224,14 @@ __wt_update_serial(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_PAGE *page
     WT_UPDATE *obsolete, *upd;
     wt_timestamp_t obsolete_timestamp, prev_upd_ts;
     uint64_t txn;
+    u_int upd_count;
 
     /* Clear references to memory we now own and must free on error. */
     upd = *updp;
     *updp = NULL;
     prev_upd_ts = WT_TS_NONE;
+    upd_count = 0;
```

Comment:
We do not need to initialize upd_count here, I will prefer this be done as a side effect of calling __wt_update_obsolete_check